### ISSUE TITLE
No support for Transform objects

### ISSUE THREAD

**Issue by tomrijndorp** [matlab2tikz/matlab2tikz#873]
Currently there is no support for Transform (`matlab.graphics.primitive.Transform`) objects, which have the ability to rotate the reference frame in which content is drawn.

    fig = figure(1); clf;
    ax = gca;

    % let's plot in a reference frame that is NED
    hgNED = hgtransform('Parent', ax);

    % Draw unit vectors
    hx = plot([0 1], [0 0], 'rx-', 'Parent', hgNED);
    hy = plot([0 0], [1 0], 'go-', 'Parent', hgNED);

    axis equal;
    xlim([-1 1]); ylim([-1 1]);
    xlabel('x');
    ylabel('y');

    hgNED.Matrix = makehgtform('zrotate', pi/2, 'xrotate', pi);  % x axis now points up

    figutils.matlab2tikz.matlab2tikz('test.tex', 'standalone', true);

MATLAB screen shot:
![screen shot 2016-03-01 at 13 46 18](https://cloud.githubusercontent.com/assets/15803419/13427398/13c92424-dfb4-11e5-8922-96b8d5b3a187.png)

Tikz output (note red and green line are swapped):
[test.pdf](https://github.com/matlab2tikz/matlab2tikz/files/152766/test.pdf)

**Comment by egeerardyn** [comment]
Thank you for this feature request.

I have little experience with those transforms, but I know that we do have support for `hgtransform` in 3D (and explicitly not in 2D plots). So, as a work-around, it might work to rotate your axes slightly: `matlab2tikz` should pick that up and 
However, I don't really see a reason why we couldn't support this in 2D as well.

Just out of curiosity: is there a specific use-case where this comes in handy?

**Comment by tomrijndorp** [comment]
Thanks for the suggestion and the extremely fast response; I'll look into it. If you already support the 3D case my bet is this should be trivial.

In terms of a use case: I've been in a perpetual struggle with getting MATLAB to plot things the way I'd like it to. The idea to use this approach came from translating some results I had produced in OpenGL code to MATLAB where a 2D representation was sufficient. I could also imagine using (2D) transforms for producing a zoomed in detail view inside the axes itself.

**Comment by egeerardyn** [comment]
@tomrijndorp I agree the implementation shouldn't be that hard (just a question of handling/faking the ZData where needed, and it removes quite some checks for 3D plots everywhere).

With respect to your zoom use-case, I think the [MagInset](http://www.mathworks.com/matlabcentral/fileexchange/49055-maginset) function does just that and it should be compatible with matlab2tikz as far as I know.
